### PR TITLE
create_model support generics model

### LINKED ISSUE #3945: create_model not support generics model
### Checks

* [x] I added a descriptive title to this issue
* [x] I have searched (google, github) for similar issues and couldn't find anything
* [x] I have read and followed [the docs](https://pydantic-docs.helpmanual.io/) and still think this is a bug

# Bug

```
from typing import Generic, TypeVar

from pydantic import Field, create_model
from pydantic.generics import GenericModel

T = TypeVar("T")


class TestGenericModel(GenericModel):
    pass


AAModel = create_model("AAModel", __base__=(TestGenericModel, Generic[T]), aa=(int, Field(0)))

AAModel[int](aa=1)
```

output
```py
Traceback (most recent call last):
  File "test.py", line 13, in <module>
    AAModel = create_model("AAModel", __base__=(TestGenericModel, Generic[T]), aa=(int, Field(0)))
  File "pydantic/main.py", line 972, in pydantic.main.create_model
TypeError: type() doesn't support MRO entry resolution; use types.new_class()
```

### PR DESCRIPTION
**Pull request by hot123s**
## Change Summary

create_model support generics model

## Related issue number

fix #3945

## Checklist

* [x] Unit tests for the changes exist
* [x] Tests pass on CI and coverage remains at 100%
* [x] Documentation reflects the changes where applicable
* [x] `changes/<pull request or issue id>-<github username>.md` file added describing change
  (see [changes/README.md](https://github.com/samuelcolvin/pydantic/blob/master/changes/README.md) for details)
* [x] My PR is ready to review, **please add a comment including the phrase "please review" to assign reviewers**

### CODE REVIEW

**Comment by hot123s** [comment]
please review

**Review by samuelcolvin — APPROVED** [review]
LGTM, but I wonder if this could have any unintended side effects?

@PrettyWood @hramezani?

**Comment by hramezani** [comment]
> but I wonder if this could have any unintended side effects?

I am not sure. but we need a rebase here to properly handle `__cls_kwargs__` and `kwds`

please update

**Comment by hot123s** [comment]
I try to extract code from `types.new_class` to avoid using lambda function

this is source code in python3.8 official library

```python
def new_class(name, bases=(), kwds=None, exec_body=None):
    resolved_bases = resolve_bases(bases)
    meta, ns, kwds = prepare_class(name, resolved_bases, kwds)
    if exec_body is not None:
        exec_body(ns)
    if resolved_bases is not bases:
        ns['__orig_bases__'] = bases
    return meta(name, resolved_bases, ns, **kwds)
```

I'm alse not sure `types.new_class` could have any unintended side effects in this project.But this is a official function to create new class, and maybe this is good alternative for `type`。

@samuelcolvin @hramezani

**Comment by hramezani** [comment]
Thanks @hot123s for the explanaitons.

Please consider you need to rebase your patch first.

**Review by PrettyWood — APPROVED** [review]
LGTM! IMO it's ok to add this in v1.10

**Comment by hot123s** [comment]
@hramezani I have rebased my branch, and compatible with `__cls_kwargs__`, and add corresponding test case.Please review it.

**Review by hramezani — APPROVED** [review]
Thanks @hot123s  :+1:

File: changes/3945-hot123s.md [hunk 1]
```diff
@@ -0,0 +1 @@
+create_model support generics model
```
Comment (samuelcolvin) [inline changes/3945-hot123s.md:1]:
```suggestion
Support generics model with `create_model`
```

**Review by samuelcolvin — APPROVED** [review]
otherwise LGTM.

**Comment by samuelcolvin** [comment]
Thanks so much.

**Comment by samuelcolvin** [comment]
See #4432 related workaround which I think is required due to this PR.

**Pull request merged**